### PR TITLE
fix(NextApiRequest) Set correct type for query and cookies

### LINKED ISSUE #22152: NextApiRequest.query has wrong type
**What version of Next.js are you using?**

v10.0.6

**What version of Node.js are you using?**

v14.15.4

**What browser are you using?**

Firefox

**What operating system are you using?**

Windows + WSL

**How are you deploying your application?**

next dev

**Describe the Bug**

The type of `NextApiRequest.query` seems to be

```
NextApiRequest.query: {
    [key: string]: string | string[];
}
```

but in reality the value can also be `undefined`.

**Expected Behavior**

The type should be

```
NextApiRequest.query: {
    [key: string]: string | string[] | undefined;
}
```

**To Reproduce**

The following application added as an API page easily shows the problem.

```
import { NextApiRequest, NextApiResponse } from "next";

export default (req: NextApiRequest, res: NextApiResponse) => {
  const { id } = req.query;

  console.log(id);

  res.status(200).json({ id });
};
```

If I'm reading correctly, the problem is [here](https://github.com/vercel/next.js/blob/56b149f7bee9e3fe0565461224a0a94def24b6d0/packages/next/next-server/lib/utils.ts#L203) and the fix would just be adding `| undefined` to the type.

### PR DESCRIPTION
**Pull request by arthurjdam**
Type updates only:

1. Make `query` for NextApiRequest consistent with its uses elsewhere (NextPageContext, NextRouter, GetServerSidePropsContext).
2. Make `cookies` show as a correct type (i.e. req.cookies.foo should be warned as `possibly undefined` by Typescript, whereas it is currently inferred as type `string`, which could in turn lead to an unexpected `object is undefined`-error).

Would close #22152

### CODE REVIEW

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall decrease ✓</summary>

|  | vercel/next.js canary  | arthurjdam/next.js issues/next-api-request-query-type | Change |
| - | - | - | - |
| buildDuration | 11s | 10.9s | -74ms |
| nodeModulesSize | 69.8 MB | 69.8 MB | -41 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | arthurjdam/next.js issues/next-api-request-query-type | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 1.998 | 1.983 | -0.01 |
| / avg req/sec | 1250.94 | 1260.71 | +9.77 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.252 | 1.236 | -0.02 |
| /error-in-render avg req/sec | 1997.33 | 2022.4 | +25.07 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | arthurjdam/next.js issues/next-api-request-query-type | Change |
| - | - | - | - |
| 677f882d2ed8..5cb0.js gzip | 13.1 kB | 13.1 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-HASH.js gzip | 6.63 kB | 6.63 kB | ✓ |
| webpack-HASH.js gzip | 751 B | 751 B | ✓ |
| Overall change | 59.5 kB | 59.5 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | arthurjdam/next.js issues/next-api-request-query-type | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31.3 kB | 31.3 kB | ✓ |
| Overall change | 31.3 kB | 31.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | arthurjdam/next.js issues/next-api-request-query-type | Change |
| - | - | - | - |
| _app-2a09aa2..4a98.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-8b758..aef6.js gzip | 3.46 kB | 3.46 kB | ✓ |
| amp-e3e9bc99..932c.js gzip | 536 B | 536 B | ✓ |
| hooks-5023ef..3670.js gzip | 888 B | 888 B | ✓ |
| index-3a2b03..c677.js gzip | 227 B | 227 B | ✓ |
| link-177412b..043e.js gzip | 1.67 kB | 1.67 kB | ✓ |
| routerDirect..1a66.js gzip | 303 B | 303 B | ✓ |
| withRouter-9..1b7d.js gzip | 302 B | 302 B | ✓ |
| Overall change | 8.66 kB | 8.66 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | arthurjdam/next.js issues/next-api-request-query-type | Change |
| - | - | - | - |
| _buildManifest.js gzip | 347 B | 347 B | ✓ |
| Overall change | 347 B | 347 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | arthurjdam/next.js issues/next-api-request-query-type | Change |
| - | - | - | - |
| index.html gzip | 614 B | 614 B | ✓ |
| link.html gzip | 621 B | 621 B | ✓ |
| withRouter.html gzip | 609 B | 609 B | ✓ |
| Overall change | 1.84 kB | 1.84 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall decrease ✓</summary>

|  | vercel/next.js canary  | arthurjdam/next.js issues/next-api-request-query-type | Change |
| - | - | - | - |
| buildDuration | 13.4s | 13s | -366ms |
| nodeModulesSize | 69.8 MB | 69.8 MB | -41 B |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | arthurjdam/next.js issues/next-api-request-query-type | Change |
| - | - | - | - |
| 677f882d2ed8..5cb0.js gzip | 13.1 kB | 13.1 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-HASH.js gzip | 6.63 kB | 6.63 kB | ✓ |
| webpack-HASH.js gzip | 751 B | 751 B | ✓ |
| Overall change | 59.5 kB | 59.5 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | arthurjdam/next.js issues/next-api-request-query-type | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31.3 kB | 31.3 kB | ✓ |
| Overall change | 31.3 kB | 31.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | arthurjdam/next.js issues/next-api-request-query-type | Change |
| - | - | - | - |
| _app-2a09aa2..4a98.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-8b758..aef6.js gzip | 3.46 kB | 3.46 kB | ✓ |
| amp-e3e9bc99..932c.js gzip | 536 B | 536 B | ✓ |
| hooks-5023ef..3670.js gzip | 888 B | 888 B | ✓ |
| index-3a2b03..c677.js gzip | 227 B | 227 B | ✓ |
| link-177412b..043e.js gzip | 1.67 kB | 1.67 kB | ✓ |
| routerDirect..1a66.js gzip | 303 B | 303 B | ✓ |
| withRouter-9..1b7d.js gzip | 302 B | 302 B | ✓ |
| Overall change | 8.66 kB | 8.66 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | arthurjdam/next.js issues/next-api-request-query-type | Change |
| - | - | - | - |
| _buildManifest.js gzip | 347 B | 347 B | ✓ |
| Overall change | 347 B | 347 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | arthurjdam/next.js issues/next-api-request-query-type | Change |
| - | - | - | - |
| _error.js | 1.01 MB | 1.01 MB | ✓ |
| 404.html | 2.67 kB | 2.67 kB | ✓ |
| amp.amp.html | 10.4 kB | 10.4 kB | ✓ |
| amp.html | 1.86 kB | 1.86 kB | ✓ |
| hooks.html | 1.92 kB | 1.92 kB | ✓ |
| index.js | 1.01 MB | 1.01 MB | ✓ |
| link.js | 1.06 MB | 1.06 MB | ✓ |
| routerDirect.js | 1.06 MB | 1.06 MB | ✓ |
| withRouter.js | 1.06 MB | 1.06 MB | ✓ |
| Overall change | 5.2 MB | 5.2 MB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Webpack 5 Mode</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall decrease ✓</summary>

|  | vercel/next.js canary  | arthurjdam/next.js issues/next-api-request-query-type | Change |
| - | - | - | - |
| buildDuration | 12.8s | 13.1s | ⚠️ +245ms |
| nodeModulesSize | 69.8 MB | 69.8 MB | -41 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | arthurjdam/next.js issues/next-api-request-query-type | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.019 | 2.037 | ⚠️ +0.02 |
| / avg req/sec | 1238.37 | 1227.44 | ⚠️ -10.93 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.243 | 1.263 | ⚠️ +0.02 |
| /error-in-render avg req/sec | 2010.53 | 1980.07 | ⚠️ -30.46 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | arthurjdam/next.js issues/next-api-request-query-type | Change |
| - | - | - | - |
| 597-d08f5fe5..d690.js gzip | 13.1 kB | 13.1 kB | ✓ |
| framework.HASH.js gzip | 39.3 kB | 39.3 kB | ✓ |
| main-HASH.js gzip | 6.58 kB | 6.58 kB | ✓ |
| webpack-HASH.js gzip | 954 B | 954 B | ✓ |
| Overall change | 59.9 kB | 59.9 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | arthurjdam/next.js issues/next-api-request-query-type | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31.1 kB | 31.1 kB | ✓ |
| Overall change | 31.1 kB | 31.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | arthurjdam/next.js issues/next-api-request-query-type | Change |
| - | - | - | - |
| _app-6e0a903..b885.js gzip | 1.26 kB | 1.26 kB | ✓ |
| _error-9e190..f9d6.js gzip | 3.38 kB | 3.38 kB | ✓ |
| amp-7fd79cc4..99ee.js gzip | 538 B | 538 B | ✓ |
| hooks-f7e199..11b9.js gzip | 904 B | 904 B | ✓ |
| index-3dc22c..ffbb.js gzip | 232 B | 232 B | ✓ |
| link-7cc9e62..ef27.js gzip | 1.66 kB | 1.66 kB | ✓ |
| routerDirect..323a.js gzip | 308 B | 308 B | ✓ |
| withRouter-a..2ef8.js gzip | 304 B | 304 B | ✓ |
| Overall change | 8.59 kB | 8.59 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | arthurjdam/next.js issues/next-api-request-query-type | Change |
| - | - | - | - |
| _buildManifest.js gzip | 322 B | 322 B | ✓ |
| Overall change | 322 B | 322 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | arthurjdam/next.js issues/next-api-request-query-type | Change |
| - | - | - | - |
| index.html gzip | 586 B | 586 B | ✓ |
| link.html gzip | 594 B | 594 B | ✓ |
| withRouter.html gzip | 580 B | 580 B | ✓ |
| Overall change | 1.76 kB | 1.76 kB | ✓ |

</details>

#### Diffs
<details>
<summary>Diff for <strong>index.html</strong></summary>

```diff
@@ -43,7 +43,7 @@
         "props": { "pageProps": {} },
         "page": "/",
         "query": {},
-        "buildId": "kKSuwC2t9uoFZ85kLZ03f",
+        "buildId": "vwuFU4A7JABR5DhQzCwzY",
         "isFallback": false,
         "gip": true
       }
@@ -77,11 +77,11 @@
       async=""
     ></script>
     <script
-      src="/_next/static/kKSuwC2t9uoFZ85kLZ03f/_buildManifest.js"
+      src="/_next/static/vwuFU4A7JABR5DhQzCwzY/_buildManifest.js"
       async=""
     ></script>
     <script
-      src="/_next/static/kKSuwC2t9uoFZ85kLZ03f/_ssgManifest.js"
+      src="/_next/static/vwuFU4A7JABR5DhQzCwzY/_ssgManifest.js"
       async=""
     ></script>
   </body>
```
</details>
<details>
<summary>Diff for <strong>link.html</strong></summary>

```diff
@@ -48,7 +48,7 @@
         "props": { "pageProps": {} },
         "page": "/link",
         "query": {},
-        "buildId": "kKSuwC2t9uoFZ85kLZ03f",
+        "buildId": "vwuFU4A7JABR5DhQzCwzY",
         "isFallback": false,
         "gip": true
       }
@@ -82,11 +82,11 @@
       async=""
     ></script>
     <script
-      src="/_next/static/kKSuwC2t9uoFZ85kLZ03f/_buildManifest.js"
+      src="/_next/static/vwuFU4A7JABR5DhQzCwzY/_buildManifest.js"
       async=""
     ></script>
     <script
-      src="/_next/static/kKSuwC2t9uoFZ85kLZ03f/_ssgManifest.js"
+      src="/_next/static/vwuFU4A7JABR5DhQzCwzY/_ssgManifest.js"
       async=""
     ></script>
   </body>
```
</details>
<details>
<summary>Diff for <strong>withRouter.html</strong></summary>

```diff
@@ -43,7 +43,7 @@
         "props": { "pageProps": {} },
         "page": "/withRouter",
         "query": {},
-        "buildId": "kKSuwC2t9uoFZ85kLZ03f",
+        "buildId": "vwuFU4A7JABR5DhQzCwzY",
         "isFallback": false,
         "gip": true
       }
@@ -77,11 +77,11 @@
       async=""
     ></script>
     <script
-      src="/_next/static/kKSuwC2t9uoFZ85kLZ03f/_buildManifest.js"
+      src="/_next/static/vwuFU4A7JABR5DhQzCwzY/_buildManifest.js"
       async=""
     ></script>
     <script
-      src="/_next/static/kKSuwC2t9uoFZ85kLZ03f/_ssgManifest.js"
+      src="/_next/static/vwuFU4A7JABR5DhQzCwzY/_ssgManifest.js"
       async=""
     ></script>
   </body>
```
</details>
</details>
Commit: 0c3dfeb6c78e85ee1028447b2a3c57983ccc6404

**Comment by pedropbazzo** [comment]
npm version error build test

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall decrease ✓</summary>

|  | vercel/next.js canary  | arthurjdam/next.js issues/next-api-request-query-type | Change |
| - | - | - | - |
| buildDuration | 12.3s | 12s | -254ms |
| nodeModulesSize | 42.7 MB | 42.7 MB | -41 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | arthurjdam/next.js issues/next-api-request-query-type | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.233 | 2.271 | ⚠️ +0.04 |
| / avg req/sec | 1119.39 | 1100.62 | ⚠️ -18.77 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.522 | 1.513 | -0.01 |
| /error-in-render avg req/sec | 1642.38 | 1652.41 | +10.03 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | arthurjdam/next.js issues/next-api-request-query-type | Change |
| - | - | - | - |
| 677f882d2ed8..e7a0.js gzip | 13.4 kB | 13.4 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-HASH.js gzip | 6.65 kB | 6.65 kB | ✓ |
| webpack-HASH.js gzip | 751 B | 751 B | ✓ |
| Overall change | 59.7 kB | 59.7 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | arthurjdam/next.js issues/next-api-request-query-type | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31.3 kB | 31.3 kB | ✓ |
| Overall change | 31.3 kB | 31.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | arthurjdam/next.js issues/next-api-request-query-type | Change |
| - | - | - | - |
| _app-2a09aa2..4a98.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-8b758..aef6.js gzip | 3.46 kB | 3.46 kB | ✓ |
| amp-e3e9bc99..932c.js gzip | 536 B | 536 B | ✓ |
| hooks-5023ef..3670.js gzip | 888 B | 888 B | ✓ |
| index-3a2b03..c677.js gzip | 227 B | 227 B | ✓ |
| link-177412b..043e.js gzip | 1.67 kB | 1.67 kB | ✓ |
| routerDirect..1a66.js gzip | 303 B | 303 B | ✓ |
| withRouter-9..1b7d.js gzip | 302 B | 302 B | ✓ |
| Overall change | 8.66 kB | 8.66 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | arthurjdam/next.js issues/next-api-request-query-type | Change |
| - | - | - | - |
| _buildManifest.js gzip | 347 B | 347 B | ✓ |
| Overall change | 347 B | 347 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | arthurjdam/next.js issues/next-api-request-query-type | Change |
| - | - | - | - |
| index.html gzip | 613 B | 613 B | ✓ |
| link.html gzip | 619 B | 619 B | ✓ |
| withRouter.html gzip | 608 B | 608 B | ✓ |
| Overall change | 1.84 kB | 1.84 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall decrease ✓</summary>

|  | vercel/next.js canary  | arthurjdam/next.js issues/next-api-request-query-type | Change |
| - | - | - | - |
| buildDuration | 14.2s | 14.1s | -136ms |
| nodeModulesSize | 42.7 MB | 42.7 MB | -41 B |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | arthurjdam/next.js issues/next-api-request-query-type | Change |
| - | - | - | - |
| 677f882d2ed8..e7a0.js gzip | 13.4 kB | 13.4 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-HASH.js gzip | 6.65 kB | 6.65 kB | ✓ |
| webpack-HASH.js gzip | 751 B | 751 B | ✓ |
| Overall change | 59.7 kB | 59.7 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | arthurjdam/next.js issues/next-api-request-query-type | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31.3 kB | 31.3 kB | ✓ |
| Overall change | 31.3 kB | 31.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | arthurjdam/next.js issues/next-api-request-query-type | Change |
| - | - | - | - |
| _app-2a09aa2..4a98.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-8b758..aef6.js gzip | 3.46 kB | 3.46 kB | ✓ |
| amp-e3e9bc99..932c.js gzip | 536 B | 536 B | ✓ |
| hooks-5023ef..3670.js gzip | 888 B | 888 B | ✓ |
| index-3a2b03..c677.js gzip | 227 B | 227 B | ✓ |
| link-177412b..043e.js gzip | 1.67 kB | 1.67 kB | ✓ |
| routerDirect..1a66.js gzip | 303 B | 303 B | ✓ |
| withRouter-9..1b7d.js gzip | 302 B | 302 B | ✓ |
| Overall change | 8.66 kB | 8.66 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | arthurjdam/next.js issues/next-api-request-query-type | Change |
| - | - | - | - |
| _buildManifest.js gzip | 347 B | 347 B | ✓ |
| Overall change | 347 B | 347 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | arthurjdam/next.js issues/next-api-request-query-type | Change |
| - | - | - | - |
| _error.js | 1.01 MB | 1.01 MB | ✓ |
| 404.html | 2.67 kB | 2.67 kB | ✓ |
| 500.html | 2.65 kB | 2.65 kB | ✓ |
| amp.amp.html | 10.4 kB | 10.4 kB | ✓ |
| amp.html | 1.86 kB | 1.86 kB | ✓ |
| hooks.html | 1.92 kB | 1.92 kB | ✓ |
| index.js | 1.01 MB | 1.01 MB | ✓ |
| link.js | 1.06 MB | 1.06 MB | ✓ |
| routerDirect.js | 1.06 MB | 1.06 MB | ✓ |
| withRouter.js | 1.06 MB | 1.06 MB | ✓ |
| Overall change | 5.21 MB | 5.21 MB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Webpack 5 Mode</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall decrease ✓</summary>

|  | vercel/next.js canary  | arthurjdam/next.js issues/next-api-request-query-type | Change |
| - | - | - | - |
| buildDuration | 14.2s | 14.3s | ⚠️ +59ms |
| nodeModulesSize | 42.7 MB | 42.7 MB | -41 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | arthurjdam/next.js issues/next-api-request-query-type | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.278 | 2.309 | ⚠️ +0.03 |
| / avg req/sec | 1097.39 | 1082.82 | ⚠️ -14.57 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.543 | 1.567 | ⚠️ +0.02 |
| /error-in-render avg req/sec | 1619.98 | 1595.47 | ⚠️ -24.51 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | arthurjdam/next.js issues/next-api-request-query-type | Change |
| - | - | - | - |
| 597-74632e79..1d47.js gzip | 13.3 kB | 13.3 kB | ✓ |
| framework.HASH.js gzip | 39.3 kB | 39.3 kB | ✓ |
| main-HASH.js gzip | 6.59 kB | 6.59 kB | ✓ |
| webpack-HASH.js gzip | 954 B | 954 B | ✓ |
| Overall change | 60.1 kB | 60.1 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | arthurjdam/next.js issues/next-api-request-query-type | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31.1 kB | 31.1 kB | ✓ |
| Overall change | 31.1 kB | 31.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | arthurjdam/next.js issues/next-api-request-query-type | Change |
| - | - | - | - |
| _app-aedc815..1421.js gzip | 1.26 kB | 1.26 kB | ✓ |
| _error-71ec2..1a96.js gzip | 3.38 kB | 3.38 kB | ✓ |
| amp-33a09cb0..6745.js gzip | 536 B | 536 B | ✓ |
| hooks-4e968a..f870.js gzip | 902 B | 902 B | ✓ |
| index-5c6845..f75c.js gzip | 230 B | 230 B | ✓ |
| link-99f0c6c..b84a.js gzip | 1.65 kB | 1.65 kB | ✓ |
| routerDirect..bb56.js gzip | 306 B | 306 B | ✓ |
| withRouter-7..2133.js gzip | 302 B | 302 B | ✓ |
| Overall change | 8.57 kB | 8.57 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | arthurjdam/next.js issues/next-api-request-query-type | Change |
| - | - | - | - |
| _buildManifest.js gzip | 326 B | 326 B | ✓ |
| Overall change | 326 B | 326 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | arthurjdam/next.js issues/next-api-request-query-type | Change |
| - | - | - | - |
| index.html gzip | 587 B | 587 B | ✓ |
| link.html gzip | 594 B | 594 B | ✓ |
| withRouter.html gzip | 582 B | 582 B | ✓ |
| Overall change | 1.76 kB | 1.76 kB | ✓ |

</details>

#### Diffs
<details>
<summary>Diff for <strong>index.html</strong></summary>

```diff
@@ -43,7 +43,7 @@
         "props": { "pageProps": {} },
         "page": "/",
         "query": {},
-        "buildId": "qfHZtDbYQKhjR0ILRsAvb",
+        "buildId": "OlR5Ndg4cXS_a0s4RttYA",
         "isFallback": false,
         "gip": true
       }
@@ -77,11 +77,11 @@
       async=""
     ></script>
     <script
-      src="/_next/static/qfHZtDbYQKhjR0ILRsAvb/_buildManifest.js"
+      src="/_next/static/OlR5Ndg4cXS_a0s4RttYA/_buildManifest.js"
       async=""
     ></script>
     <script
-      src="/_next/static/qfHZtDbYQKhjR0ILRsAvb/_ssgManifest.js"
+      src="/_next/static/OlR5Ndg4cXS_a0s4RttYA/_ssgManifest.js"
       async=""
     ></script>
   </body>
```
</details>
<details>
<summary>Diff for <strong>link.html</strong></summary>

```diff
@@ -48,7 +48,7 @@
         "props": { "pageProps": {} },
         "page": "/link",
         "query": {},
-        "buildId": "qfHZtDbYQKhjR0ILRsAvb",
+        "buildId": "OlR5Ndg4cXS_a0s4RttYA",
         "isFallback": false,
         "gip": true
       }
@@ -82,11 +82,11 @@
       async=""
     ></script>
     <script
-      src="/_next/static/qfHZtDbYQKhjR0ILRsAvb/_buildManifest.js"
+      src="/_next/static/OlR5Ndg4cXS_a0s4RttYA/_buildManifest.js"
       async=""
     ></script>
     <script
-      src="/_next/static/qfHZtDbYQKhjR0ILRsAvb/_ssgManifest.js"
+      src="/_next/static/OlR5Ndg4cXS_a0s4RttYA/_ssgManifest.js"
       async=""
     ></script>
   </body>
```
</details>
<details>
<summary>Diff for <strong>withRouter.html</strong></summary>

```diff
@@ -43,7 +43,7 @@
         "props": { "pageProps": {} },
         "page": "/withRouter",
         "query": {},
-        "buildId": "qfHZtDbYQKhjR0ILRsAvb",
+        "buildId": "OlR5Ndg4cXS_a0s4RttYA",
         "isFallback": false,
         "gip": true
       }
@@ -77,11 +77,11 @@
       async=""
     ></script>
     <script
-      src="/_next/static/qfHZtDbYQKhjR0ILRsAvb/_buildManifest.js"
+      src="/_next/static/OlR5Ndg4cXS_a0s4RttYA/_buildManifest.js"
       async=""
     ></script>
     <script
-      src="/_next/static/qfHZtDbYQKhjR0ILRsAvb/_ssgManifest.js"
+      src="/_next/static/OlR5Ndg4cXS_a0s4RttYA/_ssgManifest.js"
       async=""
     ></script>
   </body>
```
</details>
</details>
Commit: 0206dc9b1f09707eb521d265e58628f5dcb593cb

File: packages/next/next-server/lib/utils.ts [hunk 1]
```diff
@@ -201,14 +201,12 @@ export interface NextApiRequest extends IncomingMessage {
...
-  query: {
-    [key: string]: string | string[]
-  }
+  query: ParsedUrlQuery
   /**
    * Object of `cookies` from header
    */
   cookies: {
-    [key: string]: string
+    [key: string]: string | undefined
```
Comment (styfle) [inline packages/next/next-server/lib/utils.ts:209]: This should probably use `NextApiRequestCookies`

**Pull request closed without merging**